packages/trace-viewer/src/ui/uiModeTestListView.tsx

```diff
@@ -140,7 +142,10 @@ export const TestListView: React.FC<{
     dataTestId='test-tree'
     render={treeItem => {
       return <div className='hbox ui-mode-list-item'>
-        <div className='ui-mode-list-item-title' title={treeItem.title}>{treeItem.title}</div>
+        <div className='ui-mode-list-item-title' title={treeItem.title}>
+          {treeItem.title}
+          {treeItem.kind === 'case' ? treeItem.tags.map(tag => <Tag key={tag} color={tagNametoColor(tag)} onClick={e => {e.preventDefault(); setFilterText(tag);}}>{tag.slice(1)}</Tag>) : null}
```

Comment:
- Perhaps `TagView` can calculate the color itself instead of passing it from the outside?
- Why `preventDefault()`? Do you want to `stopPropagation()` and `stopImmediatePropagation()` as well?
- I think you should update the filter to add/remove tag based on whether it was in the filter already, instead of just rewriting user's carefully crafted filter to the tag value.